addons/account/models/account.py

```diff
@@ -280,6 +280,18 @@ def _check_company_consistency(self):
         if self._cr.fetchone():
             raise UserError(_("You can't change the company of your account since there are some journal items linked to it."))
 
+    @api.constrains('user_type_id')
+    def _check_user_type_id(self):
+        for account in self:
+            if account.user_type_id.type in ('receivable', 'payable'):
+                amount_of_journals = self.env['account.journal'].search_count([
+                                                        ('type', 'in', ('sale', 'purchase')),
+                                                        '|',
+                                                        ('default_credit_account_id', '=', account.id),
+                                                        ('default_debit_account_id', '=', account.id)])
+                if amount_of_journals:
+                    raise ValidationError(_("The account is already in use in a 'sale' or 'purchase' journal. This means that the account's type couldn't be 'receivable' or 'payable'."))
```

Comment:
I think this should be done directly in sql for performance reason.
What about archived journals?